integration/docker/kill_test.go

```diff
@@ -65,32 +66,37 @@ var _ = Describe("docker kill", func() {
 	})
 
 	DescribeTable("killing container",
-		func(signal syscall.Signal, expectedExitCode int) {
+		func(signal syscall.Signal, expectedExitCode int, waitForExit bool) {
 			args = []string{"--name", id, "-dt", Image, "sh", "-c"}
 
+			trapTag := "TRAP_RUNNING"
+			trapCmd := fmt.Sprintf("trap \"exit %d\" %d; echo %s", signal, signal, trapTag)
+			infiniteLoop := "while :; do sleep 1; done"
+
 			if signal > 0 {
-				args = append(args, fmt.Sprintf("trap \"exit %d\" %d ; while : ; do sleep 1; done", signal, signal))
+				args = append(args, fmt.Sprintf("%s; %s", trapCmd, infiniteLoop))
 			} else {
-				args = append(args, fmt.Sprintf("while : ; do sleep 1; done"))
+				args = append(args, infiniteLoop)
 			}
 
 			DockerRun(args...)
 
-			// we have to wait for the container workload
-			// to process the trap.
-			time.Sleep(5 * time.Second)
-			
 			if signal > 0 {
+				for {
+					logs, _ := LogsDockerContainer(id)
```

Comment:
here is ignoring error, if the container not exist and always fails this will be an infinite loop.

Reply:
Ignoring this err is on purpose, I have added a comment to explain.